cg/cli/backup.py

```diff
@@ -27,6 +34,45 @@ def backup(context: CGConfig):
     pass
 
 
+@backup.command("encrypt-flow-cells")
+@DRY_RUN
+@click.pass_obj
+def encrypt_flow_cells(context: CGConfig, dry_run: bool):
+    """Encrypt flow cells."""
+    status_db: Store = context.status_db
+    flow_cells_dir = Path(context.flow_cells_dir)
+    LOG.debug(f"Search for flow cells ready to encrypt in {flow_cells_dir}")
+    for flow_cell_dir in flow_cells_dir.iterdir():
+        if not flow_cell_dir.is_dir():
+            continue
+        LOG.debug(f"Found directory: {flow_cell_dir}")
+        try:
+            flow_cell = FlowCellDirectoryData(flow_cell_path=flow_cell_dir)
+        except FlowCellError:
+            continue
+        db_flow_cell: Optional[Flowcell] = status_db.get_flow_cell_by_name(
+            flow_cell_name=flow_cell.id
+        )
+        if db_flow_cell and db_flow_cell.has_backup:
+            LOG.debug(f"Flow cell: {flow_cell.id} is already backed-up")
+            continue
+        flow_cell_encryption_api = FlowCellEncryptionAPI(
+            binary_path=context.encryption.binary_path,
+            dry_run=dry_run,
+            encryption_dir=Path(context.backup.encrypt_dir),
+            flow_cell=flow_cell,
+            pigz_binary_path=context.pigz.binary_path,
+            slurm_api=SlurmAPI(),
+            sbatch_parameter=context.backup.slurm_flow_cell_encryption.dict(),
+            tar_api=TarAPI(binary_path=context.tar.binary_path, dry_run=dry_run),
+        )
+        is_requirement_meet, error_msg = flow_cell_encryption_api.is_encryption_possible()
+        if not is_requirement_meet:
+            LOG.debug(error_msg)
+            continue
```

Comment:
There is a suggestion below that goes with this one. Basically if any of the checks in is_encryption_possible() possible fail an error is raised that we catch here and display that error message. Maybe a bit nicer than juggling the ```Tuple bool, str```
```suggestion
        try:
            flow_cell_encryption_api.is_encryption_possible()
        except Exception as error:
            LOG.debug(error_msg)
            continue
```

Reply:
Also, why do we initiate the encryption API just to check after whether encryption is possible. Ideally we check that before we attempt encryption, right?

Reply:
Fixed